### PR TITLE
FSE: Add focus mode and top toolbar modes

### PR DESCRIPTION
**Pull request by noahtallen**
## Description
Adds top toolbar feature toggle to the site editor. Mostly copied from edit-post. (related: #22535)

While adding this, I noticed that there are a lot of small margin & style differences between edit-site and edit-post. For example, the inserter button isn't a proper square. Additionally, we are missing most of the mobile styles already coded in edit-post. I think we need to share more code between the two since the requirements for both toolbars are nearly identical. Should that kind of refactor go into the `interface` package?

## How has this been tested?
Locally in edit site

## Screenshots 
Screen Shot 2020-05-25 at 12 37 01 PM
Screen Shot 2020-05-25 at 12 37 12 PM

## Types of changes
new feature

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-site/src/components/editor/index.js [hunk 1]
```diff
@@ -69,17 +70,30 @@ function Editor( { settings: _settings } ) {
...
+			'core/edit-site'
+		);
 		return {
-			isFullscreenActive: select( 'core/edit-site' ).isFeatureActive(
-				'fullscreenMode'
-			),
+			hasFixedToolbar: isFeatureActive( 'fixedToolbar' ),
+			isFullscreenActive: isFeatureActive( 'fullscreenMode' ),
+			focusMode: isFeatureActive( 'focusMode' ),
+			deviceType: __experimentalGetPreviewDeviceType(),
```
Comment (epiqueras) [inline packages/edit-site/src/components/editor/index.js:86]: It's hard to read if these have different orders.

File: packages/edit-site/src/components/editor/index.js [hunk 2]
```diff
@@ -69,17 +70,30 @@ function Editor( { settings: _settings } ) {
...
-	const deviceType = useSelect( ( select ) => {
-		return select( 'core/edit-site' ).__experimentalGetPreviewDeviceType();
-	}, [] );
+	useEffect( () => {
+		setSettings( {
+			...settings,
+			focusMode,
+			hasFixedToolbar,
+		} );
+	}, [ focusMode, hasFixedToolbar ] );
```
Comment (epiqueras) [inline packages/edit-site/src/components/editor/index.js:96]: Why is this needed?

File: packages/edit-site/src/components/header/index.js [hunk 3]
```diff
@@ -99,14 +101,26 @@ export default function Header( {
...
-	const deviceType = useSelect( ( select ) => {
-		return select( 'core/edit-site' ).__experimentalGetPreviewDeviceType();
-	}, [] );
+	const { deviceType, hasFixedToolbar } = useSelect(
+		( select ) => ( {
+			deviceType: select(
+				'core/edit-site'
+			).__experimentalGetPreviewDeviceType(),
+			hasFixedToolbar: select( 'core/edit-site' ).isFeatureActive(
+				'fixedToolbar'
```
Comment (epiqueras) [inline packages/edit-site/src/components/header/index.js:110]: No need to call `select` twice.

**Comment by epiqueras** [comment]
>Should that kind of refactor go into the interface package?

Yes

File: packages/edit-site/src/components/editor/index.js (hunk 2 above)
Comment (noahtallen) [reply packages/edit-site/src/components/editor/index.js:96]: the block editor subscribes to settings updates via context, which is updated via the the settings state here. if we want to keep these settings in that "source of truth", then we need to do a "map props to state" sort of thing.

Alternatively, we could merge these values into the settings object in the block editor component as it renders without adding them to the settings state anywhere.

Comment (noahtallen) [reply packages/edit-site/src/components/editor/index.js]: moved it to block-editor, which makes more sense as it is a block editor setting

**Review by epiqueras — APPROVED** [review]
